File: packages/zynth-web-browser/android/ZynthWebBrowser/src/main/java/dev/zynth/webbrowser/ZynthWebBrowser.kt
package dev.zynth.webbrowser

import androidx.activity.ComponentActivity
import com.zynth.kit.runtime.ZynthRuntime

object ZynthWebBrowser {
  @JvmStatic
  fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
    runtime.installModules(listOf(WebBrowserModule(activity, runtime)))
  }
}

File: packages/zynth-web-browser/ios/ZynthWebBrowser.swift
import Foundation
import ZynthKit

@objc(ZynthWebBrowser)
public final class ZynthWebBrowser: NSObject {
  private static var module: WebBrowserModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard module == nil else {
      return
    }

    let instance = WebBrowserModule(runtime: runtime)
    runtime.installModules([instance])
    module = instance
  }

  @objc public static func cleanup() {
    module = nil
  }
}
